Vinyl/Vinyl.swift

```diff
@@ -8,11 +8,44 @@
 
 import Foundation
 
-final class Vinyl {
+struct Vinyl {
     
     let tracks: [Track]
     
     init(plastic: Plastic) {
         tracks = plastic.map(Track.init)
     }
+    
+    init(tracks: [Track]) {
+        
+        self.tracks = tracks
+    }
 }
+
+struct VinylFactory {
+    
+    static func createBadVinyl(url: NSURL, statusCode: Int, error: NSError? = nil, headers: HTTPHeaders = [:]) -> Vinyl {
+        
+        guard
+            let response = NSHTTPURLResponse(URL: url, statusCode: statusCode, HTTPVersion: nil, headerFields: headers)
+        else {
+            fatalError("We weren't able to create the Vinyl 😫")
+        }
+        
+        let track = Track(response: Response(urlResponse: response, error: error))
+        return Vinyl(tracks: [track])
+    }
+    
+    static func createValidVinyl(url: NSURL, body: NSData? = nil, headers: HTTPHeaders = [:]) -> Vinyl {
+        
+        guard
+            let response = NSHTTPURLResponse(URL: url, statusCode: 200, HTTPVersion: nil, headerFields: headers)
+        else {
+                fatalError("We weren't able to create the Vinyl 😫")
+        }
+        
+        let track = Track(response: Response(urlResponse: response, body: body))
+        return Vinyl(tracks: [track])
+    }
```

Comment:
I think we need a generic method to provide some flexibility and handle cases where the request contains a status code 4xx or 5xx but also contains a body.